src/main/java/com/epam/aidial/core/storage/BlobStorage.java

```diff
@@ -254,6 +256,20 @@ private static ContentMetadata buildContentMetadata(String contentType) {
         return BaseMutableContentMetadata.fromContentMetadata(contentMetadata);
     }
 
+    private static CredentialProvider getCredentialProvider(StorageProvider provider, String identity, String credential) {
+        return switch (provider) {
+            case S3, AZURE_BLOB, GOOGLE_CLOUD_STORAGE -> new DefaultCredentialProvider(identity, credential);
+            case FILESYSTEM -> new DefaultCredentialProvider("identity", "credential");
```

Comment:
This case could be merged into the first one.

Reply:
actually not, because identity and credential hardcoded for filesystem installation

Reply:
I believe the provider doesn't validate them. Right?
If so we can pass whatever we want.